--- quickjs/quickjs.c	2023-03-22 14:23:45.000000000 +0100
+++ quickjs/quickjs.c	2023-03-22 14:23:47.000000000 +0100
@@ -16314,6 +16314,13 @@
     sf->prev_frame = rt->current_stack_frame;
     rt->current_stack_frame = sf;
     ctx = b->realm; /* set the current realm */
+
+    // function invocation time
+    const char *name;
+    name = get_func_name(ctx, sf->cur_func);
+    struct timespec start, end;
+    uint64_t duration;
+    clock_gettime(CLOCK_MONOTONIC, &start);
     
  restart:
     for(;;) {
@@ -18707,6 +18713,13 @@
     }
     rt->current_stack_frame = sf->prev_frame;
+
+    // function invocation time
+    clock_gettime(CLOCK_MONOTONIC, &end);
+    duration = ((uint64_t)end.tv_sec * 1000000ULL + (end.tv_nsec / 1000LLU))-((uint64_t)start.tv_sec * 1000000ULL + (start.tv_nsec / 1000LLU));
+    if (duration > 0 && duration < 30*1000*1000 && name && name[0] != '\0') {
+        printf(":::invocation,%s,%llu\n", name, duration);
+    }
+
     return ret_val;
 }
 
